packages/sdk/src/dataProtector/sharing/createCollection.ts

```diff
@@ -0,0 +1,42 @@
+import { ethers, type Contract } from 'ethers';
+import { ABI as sharingABI } from '../../contracts/sharingAbi.js';
+import { WorkflowError } from '../../utils/errors.js';
+import { throwIfMissing } from '../../utils/validators.js';
+import { AddressOrENS, IExecConsumer } from '../types.js';
+
+export type CreateCollectionResponse = {
+  collectionId: AddressOrENS;
+};
+
+export const createCollection = async ({
+  iexec = throwIfMissing(),
+  sharingContractAddress,
+}: IExecConsumer & {
+  sharingContractAddress: AddressOrENS;
+}): Promise<CreateCollectionResponse> => {
+  const { provider, signer } = await iexec.config.resolveContractsClient();
+
+  const sharingContract = new ethers.Contract(
+    sharingContractAddress,
+    sharingABI,
+    provider
+  );
+
+  const transactionReceipt = await (sharingContract.connect(signer) as Contract)
+    .createCollection()
+    .then((tx) => tx.wait())
+    .catch((e: Error) => {
+      throw new WorkflowError(
+        'Failed to create collection into sharing smart contract',
+        e
+      );
+    });
+
+  const mintedTokenId = transactionReceipt.logs.find(
+    ({ eventName }) => eventName === 'Transfer'
+  )?.args[2];
+
+  return {
+    collectionId: mintedTokenId,
+  };
```

Comment:
As tokenid is an uint256 in the ABI `mintedTokenId` is going to be a bigint. Assuming we are using a sequence for collectionIds, we can safely cast it into a number that will fit into safe integers.

```suggestion
  const mintedTokenId = transactionReceipt.logs.find(
    ({ eventName }) => eventName === 'Transfer'
  )?.args[2];

  return {
    collectionId: Number(mintedTokenId),
  };
```

Reply:
Would it make sense to add `as bigint` line 37 to better understand the Number casting line 40?

Reply:
yes! It will make it clearer